From f385f64aeb3c7d6601f725c1d3b854b59f7dfba2 Mon Sep 17 00:00:00 2001
From: saravananm <saravananm@ami.com>
Date: Fri, 29 Sep 2023 13:07:23 +0530
Subject: [PATCH] Fix-for-GetSMTP-ConfigParam5

Description :
Get SMTP configuration parameter 5 Receiver Mail address with empty block 1 not showing any response code while getting,So,Add the Condition to Response Code.

Tested:

Build Successfully and Verified,its Working As Expected.

Before Fix:
msgubuntu4@msgubuntu4-OptiPlex-5080:~$  ipmitool -H 10.0.100.120 -U root -P 0penBmc -I lanplus -C 17 raw 0x30 0xE4 0x05 0x01

msgubuntu4@msgubuntu4-OptiPlex-5080:~$

After Fix:
msgubuntu4@msgubuntu4-OptiPlex-5080:~$  ipmitool -H 10.0.100.120 -U root -P 0penBmc -I lanplus -C 17 raw 0x30 0xE4 0x05 0x01

Unable to send RAW command (channel=0x0 netfn=0x0 lun=0x0 cmd=0x8 rsp=0xce): Response Error

Working As Expected.

Signed-off-by: saravananm <saravananm@ami.com>
---
 src/oemcommands.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/oemcommands.cpp b/src/oemcommands.cpp
index 85ef0e0..a1fc660 100644
--- a/src/oemcommands.cpp
+++ b/src/oemcommands.cpp
@@ -4253,7 +4253,12 @@ ipmi::RspType<std::vector<uint8_t>> ipmiOEMGetSmtpConfig(ipmi::Context::ptr ctx,
             {
                 return responseReqDataLenInvalid();
             }
-            resData = convertToBytes(rec[index - 1]);
+            std::string str = rec[index - 1];
+            if (str.empty())
+            {
+                return ipmi::responseResponseError();
+            }
+            resData = convertToBytes(str);
             break;
         }
         default:
-- 
2.25.1

